### PR TITLE
Rough outline of search by file upload functionality

### PR DESCRIPTION
**Pull request by jorntx**
A very rough outline of the functionality (and sorry for the crude css but I think we should fix this in a major search box overhaul).

Take a look and see if you like where this is heading. If yes we should merge it and make some tickets for improving it.

### CODE REVIEW

File: src/js/src/components/SearchUploadFile.vue [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  uploadFile(){
+     //TODO spinner
+     requestService.uploadFileRequest(this.fileToUpload[0])
+        .then(response => {
+          //Search for file
+         this.updateQuery(this.createRequestQuery(response.data))
+         this.requestSearch({query:this.createRequestQuery(response.data), facets:this.searchAppliedFacets})
+        })
+        .catch(() => {
+           //TODO Call error handler - something went wrong
```
Comment (jesperlauridsen) [inline src/js/src/components/SearchUploadFile.vue:51]: Gotta start keeping track of our error handling places

File: src/js/src/components/SearchUploadFile.vue [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      this.fileToUpload = this.$refs.file.files
+    },
+
+  uploadFile(){
+     //TODO spinner
+     requestService.uploadFileRequest(this.fileToUpload[0])
+        .then(response => {
+          //Search for file
+         this.updateQuery(this.createRequestQuery(response.data))
+         this.requestSearch({query:this.createRequestQuery(response.data), facets:this.searchAppliedFacets})
```
Comment (jesperlauridsen) [inline src/js/src/components/SearchUploadFile.vue:48]: very good to update the query here aswell - the requestSearch method is being expanded in another incoming branch where you'll need the solrSettings param to be passed to the search function aswell. Just a heads up.

File: src/js/src/components/SearchBox.vue [hunk 3]
```diff
@@ -14,23 +14,31 @@
...
               title="Clear search and results"
               type="button"
-              @click="clearResultsAndSearch" />     
+              @click="clearResultsAndSearch" />
     </form>
+    <div @click="showUploadFileSearch = !showUploadFileSearch">
+      <span class="searchBoxActionLink">Search with uploaded file</span>
+    </div>
     <applied-search-facets />
+    <search-upload-file v-if="showUploadFileSearch" />
```
Comment (jesperlauridsen) [inline src/js/src/components/SearchBox.vue:23]: The placement of the button is a little weird if you start a search and place some filters. Perhaps we could find a better way to start the interaction - maybe clear the search all together and push a new state when you want to search for an uploaded file?

**Review by jesperlauridsen — APPROVED** [review]
All in all very nice - the functionality is solid. the UI placement could imho be worked in a little better - but theres lots of time to find out how we're gonna have all the functionality play together nicely. Approved from here! :+1:

File: src/js/src/components/SearchUploadFile.vue (hunk 1 above)
Comment (jorntx) [reply src/js/src/components/SearchUploadFile.vue:51]: Yes - we need to put this task on the board

File: src/js/src/components/SearchUploadFile.vue (hunk 2 above)
Comment (jorntx) [reply src/js/src/components/SearchUploadFile.vue:48]: Thanks for the heads up!

File: src/js/src/components/SearchBox.vue (hunk 3 above)
Comment (jorntx) [reply src/js/src/components/SearchBox.vue:23]: I totally agree here and actually I forgot about testing with filters applied :) I think you are spot on with the interaction - we should consider removing or making the search box  (appear) inactive when searching with uploaded file.

**Comment by jorntx** [comment]
Thanks for the feedback - I converted your comments to tickets

**Pull request merged**